Picture-In-Picture/Lets-Build-OTPublisher/ExampleVideoRender.swift

```diff
@@ -0,0 +1,265 @@
+    fileprivate func calculatePlaneSize(forFrame frame: OTVideoFrame)
+        -> (ySize: Int, uSize: Int, vSize: Int)
+    {
+        guard let frameFormat = frame.format
+            else {
+                return (0, 0 ,0)
+        }
+        let baseSize = Int(frameFormat.imageWidth * frameFormat.imageHeight) * MemoryLayout<GLubyte>.size
+        return (baseSize, baseSize / 4, baseSize / 4)
+    }
+}
+
+extension ExampleVideoRender: OTVideoRender {
+    func renderVideoFrame(_ frame: OTVideoFrame) {
+        if let format = frame.format {
+            frameLock.lock()
+            assert(format.pixelFormat == .I420)
+            
+            if let sampleBuffer = createSampleBufferWithVideoFrame(frame,
+                                                                   width: Int(frame.format!.imageWidth),
+                                                                   height: Int(frame.format!.imageHeight)) {
+                bufferDisplayLayer.enqueue(sampleBuffer)
+                pipBufferDisplayLayer?.enqueue(sampleBuffer)
+            }
+            
+            frameLock.unlock()
+        }
+    }
+    
+    
+    func createSampleBufferWithVideoFrame(_ frame: OTVideoFrame, width: Int, height: Int) -> CMSampleBuffer? {
+        
+        let pixelAttributes: NSDictionary = [kCVPixelBufferIOSurfacePropertiesKey as String: [:]]
+        var pixelBuffer: CVPixelBuffer?
+        let result = CVPixelBufferCreate(kCFAllocatorDefault, width, height, kCVPixelFormatType_32BGRA, pixelAttributes as CFDictionary, &pixelBuffer)
+        
+        guard result == 0 else {
+            return nil
+        }
+        _ = accel.convertFrameVImageYUV(frame, to: pixelBuffer)
+        let s = createSampleBufferFrom(pixelBuffer: pixelBuffer!)
+        
+        
+        return s
+    }
+    
+    func createSampleBufferFrom(pixelBuffer: CVPixelBuffer) -> CMSampleBuffer? {
+        CVPixelBufferLockBaseAddress(pixelBuffer, .readOnly)
+        
+        var sampleBuffer: CMSampleBuffer?
+        
+        
+        let now = CMTimeMakeWithSeconds(CACurrentMediaTime(), preferredTimescale: 1000)
+        var timingInfo = CMSampleTimingInfo(duration: CMTimeMakeWithSeconds(1, preferredTimescale: 1000), presentationTimeStamp: now, decodeTimeStamp: now)
+        var formatDescription: CMFormatDescription? = nil
+        CMVideoFormatDescriptionCreateForImageBuffer(allocator: kCFAllocatorDefault, imageBuffer: pixelBuffer, formatDescriptionOut: &formatDescription)
+        
+        let osStatus = CMSampleBufferCreateReadyWithImageBuffer(
+            allocator: kCFAllocatorDefault,
+            imageBuffer: pixelBuffer,
+            formatDescription: formatDescription!,
+            sampleTiming: &timingInfo,
+            sampleBufferOut: &sampleBuffer
+        )
+        
+        //        print("osStatus CMSampleBufferCreateReadyWithImageBuffer = \(osStatus)")
+        // Print out errors
+        if osStatus == kCMSampleBufferError_AllocationFailed {
```

Comment:
For escalability reasons, I would move all these errors into a map and just do something like this:

if let errorMessage = errorMessages[osStatus] {
    print("osStatus == \(errorMessage)")
}